src/DataValues/DecimalValue.php

```diff
@@ -63,7 +63,7 @@ public function __construct( $value ) {
 			throw new IllegalValueException( 'Value must be at most 127 characters long.' );
 		}
 		if ( !preg_match( self::QUANTITY_VALUE_PATTERN, $value ) ) {
-			throw new IllegalValueException( 'Value must match the pattern for decimal values.' );
+			throw new IllegalValueException( 'Value must match the pattern for decimal values. Malformed: ' . $value );
```

Comment:
Weird technical wording. Suggestion: `"\"$value\" is not a well formed decimal value"`.